src/plugins/discover/public/application/components/utils/page_size_options.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Generates an array of pagination options based on the provided `stepSize`.
+ * The array includes default values (25, 50, 100) and additional options derived from the `stepSize` setting.
+ * Ensures uniqueness and sorts the array in ascending order, representing available page size options for pagination.
+ * @param {number} stepSize - The sample size used to determine additional pagination options.
+ * @returns {number[]} - An array of available page size options.
+ */
+
+export const generatePageSizeOptions = (stepSize: number): number[] => {
```

Comment:
I think it should be called `pageSizeLimit` here: 
```suggestion
export const generatePageSizeOptions = (pageSizeLimit: number): number[] => {
```

And  `const stepSize = 500;` on line 18 :)

Reply:
Ohh, I changed the wrong one.